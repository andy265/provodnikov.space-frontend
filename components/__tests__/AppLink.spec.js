const { createRenderer } = require('vue-server-renderer')
const { shallow } = require('vue-test-utils')
const AppLink = require('~/components/AppLink')

describe('AppLink.vue', () => {
  it('has same HTML structure', () => {
    let wrapper = shallow(AppLink, {
      propsData: {
        to: '/page'
      },
      slots: {
        default: '<span>some text</span>'
      }
    })
    let renderer = createRenderer()
    // eslint-disable-next-line
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('can open site in new tab', () => {
    let wrapper = shallow(AppLink, {
      propsData: {
        to: '/page',
        isBlank: true
      }
    })
    expect(wrapper.find('a').hasAttribute('target', '_blank')).toBeTruthy()
  })
})
