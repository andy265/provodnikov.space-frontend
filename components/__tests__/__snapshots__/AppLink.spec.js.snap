// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLink.vue has same HTML structure 1`] = `<a href="/page" target="_self" rel="noopener noreferrer" class="app-link"><span>some text</span></a>`;
